aspnetcore/release-notes/aspnetcore-7.0.md

```diff
@@ -209,6 +209,266 @@ SignalR hub methods now support injecting services through dependency injection
+
+### `System.Security.Cryptography` support on WebAssembly
+
+.NET 6 supported the SHA family of hashing algorithms when running on WebAssembly. .NET 7 enables more cryptographic algorithms by taking advantage of [:::no-loc text="SubtleCrypto":::](https://developer.mozilla.org/docs/Web/API/SubtleCrypto), when possible, and falling back to a .NET implementation when :::no-loc text="SubtleCrypto"::: can't be used. The following algorithms are supported on WebAssembly in .NET 7:
+
+* SHA1
+* SHA256
+* SHA384
+* SHA512
+* HMACSHA1
+* HMACSHA256
+* HMACSHA384
+* HMACSHA512
+* AES-CBC
+* PBKDF2
+* HKDF
+
+For more information, see [Developers targeting browser-wasm can use Web Crypto APIs (dotnet/runtime #40074)](https://github.com/dotnet/runtime/issues/40074).
+
+<!-- 
+
+    HOLD
+
+    🪲 fix ... not addressed in docs
+
+### Generic Blazor component can't correctly infer it's type when type has an additional constraint
+
+https://github.com/dotnet/aspnetcore/issues/25588
+
+-->
+
+<!--
+
+    HOLD
+
+    🪲 fix ... not addressed in docs
+
+### `CascadingTypeParameter` problem
+
+https://github.com/dotnet/aspnetcore/issues/38479
+
+-->
+
+### Inject services into custom validation attributes
+
+You can now inject services into custom validation attributes. Blazor sets up the `ValidationContext` so that it can be used as a service provider.
+
+For more information, see <xref:blazor/forms-and-input-components?view=aspnetcore-7.0#custom-validation-attributes>.
+
+### `Input*` components outside of an `EditContext`/`EditForm`
+
+The built-in input components are now supported outside of a form in Razor component markup.
+
+For more information, see <xref:blazor/forms-and-input-components?view=aspnetcore-7.0#built-in-input-components>.
+
+### Project template changes
+
+When .NET 6 was released last year, the HTML markup of the `_Host` page (`Pages/_Host.chstml`) was split between the `_Host` page and a new `_Layout` page (`Pages/_Layout.chstml`) in the .NET 6 Blazor Server project template.
+
+In .NET 7, the HTML markup has been recombined with the `_Host` page in project templates.
+
+Several additional changes were made to the Blazor project templates. It isn't feasible to list every change to the templates in the documentation. To migrate an app to .NET 7 in order to adopt all of the changes, see <xref:migration/60-to-70#blazor>.
+
+### Experimental `QuickGrid` component
+
+The new `QuickGrid` component provides a convenient data grid component for most common requirements and as a reference architecture and performance baseline for anyone building Blazor data grid components.
+
+For more information, see <xref:blazor/components/index?view=aspnetcore-7.0#quickgrid-component>.
+
+Live demo: [QuickGrid for Blazor sample app](https://aspnet.github.io/quickgridsamples/)
+
+<!--
+
+    HOLD
+
+    It looks like this is an impl detail that only pertains to browser nav history state.
+
+### `NavigationManager` supports state
```

Comment:
We have new public API that lets you manipulate and access the browser history state, so this should be included. See https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-7-preview-7/#pass-state-using-navigationmanager

Reply:
Sorry ... I didn't recognize it. It's listed in another entry. I'll work on this one a bit more.

It's in the docs at ...

https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-7.0#navigation-options